### PR TITLE
5 get article comments - completed challenge 6 - GET /api/article/:article_id/comments

### PR DESCRIPTION
**Pull request by Jookz**

### CODE REVIEW

File: __tests__/nc_news.test.js [hunk 1]
```diff
@@ -86,3 +86,44 @@ describe('GET /api/articles/:article_id', () => {
...
 });
+
+describe('GET /api/articles/:article_id/comments', () => {
+    it('GET:200 should return 200 status code', () => {
+        return request(app).get("/api/articles/1/comments").expect(200);
+    });
+    it('GET:200 should provide all comments for specified article in correct format', () => {
+        return request(app).get("/api/articles/1/comments").then(({body}) => {
+            expect(body).toHaveLength(11);
+            console.log(body)
```
Comment (B7Bprog) [inline __tests__/nc_news.test.js:97]: Please remove all debugging logs.

File: __tests__/nc_news.test.js [hunk 2]
```diff
@@ -86,3 +86,44 @@ describe('GET /api/articles/:article_id', () => {
...
+                expect(comment).toHaveProperty("body");
+                expect(comment).toHaveProperty("article_id");
+            })
+        })
+    });
+    it('GET:200 should return most recent comments first', () => {
+        return request(app).get("/api/articles/1/comments").then(({body}) => {
+            expect(body).toBeSortedBy("created_at", {descending: true})
+        })
+    });
```
Comment (B7Bprog) [inline __tests__/nc_news.test.js:112]: Please add another test for a successful response where the article exists but there are no comments for it. Responding with an empty array.

File: controllers/topics.controller.js [hunk 3]
```diff
@@ -23,4 +23,14 @@ exports.getEndpoints = (req, res, next) => {
     res.status(200).send({endpoints});
 }
 
+exports.getComments = (req, res, next) => {
+    const article_id = req.params.article_id;
+    fetchComments(article_id).then(({rows}) => {
```
Comment (B7Bprog) [inline controllers/topics.controller.js:28]: You will need to check the reason for the query returning empty rows:
- article exists but has no comments.
- article does not exist.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Jonny, please check my comments and add those tests/refactor code, then please send this PR back to use with the updates.

File: controllers/topics.controller.js [hunk 4]
```diff
@@ -28,4 +28,22 @@ exports.getArticles = (req, res, next) => {
...
 
+exports.getComments = (req, res, next) => {
+    const article_id = req.params.article_id;
+    const promises = [fetchComments(article_id)];
+
+    
+    if(article_id) {
+        promises.push(fetchArticleId(article_id));
+    }
+
```
Comment (NRMSMITH) [inline controllers/topics.controller.js:39]: Is this conditional logic needed? The article_id should be there - if not it won't get to this endpoint :)

**Review by NRMSMITH — APPROVED** [review]
I've asked a q about your conditional logic - please address this and then merge :)

**Pull request merged**